.github/workflows/build.yml

```
@@ -23,18 +23,21 @@ jobs:
         with:
           cache-read-only: ${{ github.event_name == 'pull_request' }}
 
-      - name: Build & Publish Snapshot
+      - name: Build & Publish to Maven Local
         shell: bash
-        env:
-          MAVEN_SNAPSHOTS_USERNAME: ${{ secrets.MAVEN_USERNAME }}
-          MAVEN_SNAPSHOTS_PASSWORD: ${{ secrets.MAVEN_PASSWORD }}
-        run: |
-          chmod +x ./gradlew
-          ./gradlew --stacktrace :publishToMavenLocal :publishAllPublicationsToAliucordSnapshotsRepository -PVERSION_NAME=${GITHUB_SHA::7}
+        run: ./gradlew --stacktrace :publishToMavenLocal
 
       - name: Upload artifacts
         uses: actions/upload-artifact@v4
         with:
           name: gradle-plugin
           path: ~/.m2/repository/**
           if-no-files-found: error
+
+      - name: Publish Snapshot
+        shell: bash
+        env:
+          MAVEN_SNAPSHOTS_USERNAME: ${{ secrets.MAVEN_USERNAME }}
+          MAVEN_SNAPSHOTS_PASSWORD: ${{ secrets.MAVEN_PASSWORD }}
+        if: ${{ env.MAVEN_SNAPSHOTS_USERNAME != '' && env.MAVEN_SNAPSHOTS_PASSWORD != '' }}
```

1. I believe we should check that this isn't a pr run as well, just to be sure
2. I don't think `env` properties exist here, you would need to write to `env` with something like:
```bash
"NEED=false" >> "$GITHUB_ENV"
```
in an earlier step for this to be accessible. Using the secrets directly should work instead.

```suggestion
        if: ${{ github.event_name != 'pull_request' && secrets.MAVEN_SNAPSHOTS_USERNAME != '' && secrets.MAVEN_SNAPSHOTS_PASSWORD != '' }}
```